utils/fix_cmocka_xml.py

```diff
@@ -22,22 +22,21 @@
   portions thereof marked with this legend must also reproduce the markings.
 
 '''
-import subprocess
 import os
 
 # Some fixes for the CMOCKA headers
 xml_header ="<?xml version=\"1.0\" encoding=\"UTF-8\" ?>\n"
 parent_header = "<testsuites>\n"
 parent_footer = "</testsuites>\n"
-path = os.path.join(os.path.dirname(os.path.realpath(__file__)),"../test_results/*.xml")
-cmd = "ls {}".format(path)
-out = subprocess.check_output(cmd,shell=True)
+file_extensions = "*.xml"
+path = os.path.join(os.path.dirname(os.path.realpath(__file__)),"../test_results/")
```

Comment:
Really shouldn't mix `os.path.join()` with `/` in the same construct.  The `/`'s negate the use of `os.path.join()`.

Also, should the path of the `test_results/` dir be relative to the program creating it or relative to where the program is being run (i.e. `$PWD`)?  The latter feels more flexible.  It allows for the former but allows for alternatives whereas the former really restricts where the `test_results/` dir can be.  It also seems not-so-great to be putting data into the dir where the program is (which is admittedly what ftest.sh is doing with the avocado/ results dir, but that's an abomination that needs to be fixed)

Reply:
added code:
pwd = os.getcwd()
path = os.path.join(pwd,"..","test_results") + os.path.sep